libraries/cms/application/cms.php

```diff
@@ -114,103 +114,15 @@ public function __construct(JInput $input = null, Registry $config = null, JAppl
 		}
 
 		// Enable sessions by default.
-		if (is_null($this->config->get('session')))
+		if (is_null($this->getContainer()->get('config')->get('session')))
```

Comment:
Since the application classes themselves support a configuration, I've been using the application API to inject my config object on bootup and read it during the application cycle.  Only B/C break to `JFactory::getConfig()` is that you can't retrieve the entire configuration object outside the application scope.  I guess that still leaves figuring out how to access the config outside the application on the table though.